apps/src/blockly/addons/cdoXml.js

```diff
@@ -34,22 +36,20 @@ export default function initializeBlocklyXml(blocklyWrapper) {
     return block;
   };
 
-  blocklyWrapper.Xml.domToBlockSpace = function (blockSpace, xml) {
+  // Decode an XML DOM and create blocks on the workspace.
+  blocklyWrapper.Xml.domToBlockSpace = function (workspace, xml) {
+    Blockly.Xml.createBlockOrderMap(xml);
+    const sortedBlockElements = getSortedBlockElements(
+      xml,
+      PROCEDURE_DEFINITION_TYPES
+    );
+    const blocks = [];
     // To position the blocks, we first render them all to the Block Space
     //  and parse any X or Y coordinates set in the XML. Then, we store
     //  the rendered blocks and the coordinates in an array so that we can
-    //  position them in two passes.
-    //  In the first pass, we position the visible blocks. In the second
-    //  pass, we position the invisible blocks. We do this so that
-    //  invisible blocks don't cause the visible blocks to flow
-    //  differently, which could leave gaps between the visible blocks.
-    const blocks = [];
-    xml.childNodes.forEach(xmlChild => {
-      if (xmlChild.nodeName.toLowerCase() !== 'block') {
-        // skip non-block xml elements
```

Comment:
Can you clarify why non-block XML elements are not something we need to worry about anymore?

Reply:
Answered my own question! https://github.com/code-dot-org/code-dot-org/pull/52932/files#diff-1deb52d0698ae48e74781e441207cbeb1ffffd69fc70293a3dce035cb216e6aeR82